R/fitting-and-postprocessing.R

```diff
@@ -33,19 +20,11 @@ extract_lognormal_draws <- function(data, id_vars, from_dt = FALSE) {
   draws <- posterior::as_draws_df(draws) |>
     data.table::as.data.table()
 
-  data.table::setnames(
-    draws, c("Intercept", "Intercept_sigma"), c("meanlog", "sdlog_log"),
-    skip_absent = TRUE
-  )
-
-  data.table::setnames(
-    draws, c("b_Intercept", "b_sigma_Intercept"), c("meanlog", "sdlog_log"),
-    skip_absent = TRUE
-  )
-
-  draws <- draws[, sdlog := exp(sdlog_log)]
-  draws <- draws[, list(meanlog, sdlog)]
-  draws <- add_natural_scale_mean_sd(draws)
+  draws <- draws[, mu := Intercept]
+  draws <- draws[, sigma := exp(Intercept_sigma)]
+  draws <- draws[, list(mu, sigma)]
+  class(draws) <- c(class(draws), "lognormal_samples")
+  draws <- add_mean_sd(draws)
```

Comment:
we are using this inside a function called lognormal and so we could just dispatch on the specific version i.e add_mean_sd.lognormal_samples vs the approach taken here.  As this extraction function isn't really log normal specific apart for this I guess it doesn't matter but its not ideal

Reply:
can be a new issue to refine this as I think we are likely to drop/refactor/rename this function

Reply:
Yeah I think we will change this function. Will make a new issue